### PR TITLE
Add documentation for the MySQL image

### PR DESCRIPTION
**Pull request by mnagy**

### CODE REVIEW

**Comment by soltysh** [comment]
I'd propose moving `overview.adoc` and `mysql.adoc` to `others` subdir. And copy `openshift_sti_images` into this dir under name `sti`, so the final result should look like this:
```
- openshift_images/
  - sti/
    nodejs.adoc
    overview.adoc
    ruby.adoc
  - other/
    mysql.adoc
    overview.adoc
```

File: openshift_images/mysql.adoc [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+Our OpenShift MySQL database image https://github.com/openshift/mysql[repository] contains the sources and Dockerfiles for building MySQL database images. The resulting images can be run by Docker.
```
Comment (bparees) [inline openshift_images/mysql.adoc]: s/Our/The/

File: openshift_images/mysql.adoc [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+== Versions
+Current supported versions of MySQL:
```
Comment (bparees) [inline openshift_images/mysql.adoc]: Currently

File: openshift_images/mysql.adoc [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+* https://github.com/openshift/mysql/tree/master/5.5[5.5]
+
+== Base Images
+
+* RHEL-7
+* CentOS-7
+
+[NOTE]
+====
+RHEL-7 image is not available as trusted build in https://index.docker.io[Docker Index]. In order to perform build or test action on a RHEL-7 based MySQL image, you need to run the build on properly subscribed RHEL-7 machine.
```
Comment (bparees) [inline openshift_images/mysql.adoc]: nothing is available as a trusted build in docker hub anymore, so just say that it's not available on docker hub.

File: openshift_images/mysql.adoc [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+====
+
+== Installation
+To build MySQL image, choose between CentOS7 or RHEL7 base image:
```
Comment (bparees) [inline openshift_images/mysql.adoc]: choose between *the* CentOS7 or RHEL7.....

File: openshift_images/mysql.adoc [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+----
+$ sti build https://github.com/ryanj/node-echo.git openshift/nodejs-010-rhel7 nodejs-echo-app
+$ docker run -p 3000:3000 nodejs-echo-app
+----
+
+=== CentOS-7 image
+
+----
+$ sti build https://github.com/ryanj/node-echo.git openshift/nodejs-010-centos7 nodejs-echo-app
```
Comment (bparees) [inline openshift_images/mysql.adoc]: this should be the ruby app, right?  and port 8080

File: openshift_images/mysql.adoc [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+$ docker run -p 3000:3000 nodejs-echo-app
+----
+
+To access and test if the application is running run:
+----
+$ curl 127.0.0.1:3000
+----
+
+== Test
+OpenShift https://github.com/openshift/sti-nodejs/tree/master/0.10[STI-NodeJS] repository also provides STI test framework, which launches test to check functionality of a simple nodejs application built on top of sti-nodejs image.
```
Comment (bparees) [inline openshift_images/mysql.adoc]: looks like you copy+pasted this from the nodejs repo w/o fixing the names.

File: openshift_images/mysql.adoc [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+* *MYSQL_USER* - This variable specifies username for database user that
+    will be created for your application to use.
+
+* *MYSQL_PASSWORD* - Password for the *MYSQL_USER*.
+
+* *MYSQL_DATABASE* - Name of the database that will be created. *MYSQL_USER*
+    will have full rights to the database.
+
+* *MYSQL_ROOT_PASSWORD* - Password for the root user. This variable is
+    optional. If not set, root user will not have any password set.
```
Comment (bparees) [inline openshift_images/mysql.adoc]: again we need to doc the exact behavior here, once you figure it out.

File: openshift_images/mysql.adoc (hunk 6 above)
Comment (mnagy) [inline openshift_images/mysql.adoc]: I used nodejs as a template, forgot to remove this whole section.

**Comment by adellape** [comment]
@CowboysFan Please review the proposed `build_cfg` changes (also discussed in above comments).

**Comment by CowboysFan** [comment]
@soltysh We're trying to avoid subdirectories, and I don't even think our build system will allow that. Both of these topic dirs and the topics within need quite a bit of restructuring in my opinion.

@adellape Let's go ahead and merge this and we'll take a look at it afterwards, taking suggestions from @soltysh into consideration. I'm working on the restructure so I'll just make it a part of that process. Thanks.

**Comment by bparees** [comment]
@CowboysFan @adellape don't merge this yet, i made a bunch of comments that have not been addressed.

@mnagy ETA for updating this?

**Comment by mnagy** [comment]
Sorry for the tardiness. PTAL.

I addressed the comments, but kept _build_cfg.yml as it was. I removed mentions of the test target, since I decided it probably should be a developer thing that's not in the openshift docs.

**Comment by adellape** [comment]
Thanks, @mnagy. Merging.

FYI @tpoitras (per your related https://trello.com/c/jPzgJkic) and @CowboysFan (per `_build_cfg` changes for your restructure).

File: openshift_images/mysql.adoc [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+=== RHEL-7 base image
+
+----
+$ git clone https://github.com/openshift/mysql.git
+$ cd mysql
+$ make build TARGET=rhel7 VERSION=5.5
+----
+
+=== CentOS-7 base image
+
```
Comment (bparees) [inline openshift_images/mysql.adoc]: or you can just pull the centos image from docker, no need to build it.

File: openshift_images/mysql.adoc [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+$ make build VERSION=5.5
+----
+
+[NOTE]
+====
+By omitting the `VERSION` parameter, the build action will be performed on all the supported versions of MySQL. Since we are now supporting only version `5.5`, you can omit this parameter.
+====
+
+== Usage
+To create a container running MySQL, you need to is run following commands.
```
Comment (bparees) [inline openshift_images/mysql.adoc]: s/is//

**Comment by bparees** [comment]
I realize i'm post-merge here but i did make 2 minor comments, and we still need to be clearer about what happens if no admin password is set.

**Comment by adellape** [comment]
@bparees That's cool. @tpoitras will likely be making a follow-up PR for this topic per his related docs card (https://trello.com/c/jPzgJkic) and should be able to address them.

**Pull request merged**